index.d.ts

```diff
@@ -1,53 +1,21 @@
 // Type definitions for Sequelize 3.x
 // Project: http://sequelizejs.com
 
-// import * as DataTypes from './lib/data-types';
-
 import SequelizePromise = require('./lib/promise');
-import * as LibDataTypes from './lib/data-types';
+import {DataType, DataTypes as LibDataTypes} from './lib/data-types';
 
 import Utils = require('./lib/utils');
 import {Validator} from './lib/utils/validator-extras';
 
 declare module sequelize {
-  type Promise<T> = SequelizePromise<T>;
-  type DataTypes = LibDataTypes.DataTypes;
-
-  type ABSTRACT = LibDataTypes.ABSTRACT;
-  type STRING = LibDataTypes.STRING;
-  type CHAR = LibDataTypes.CHAR;
-  type TEXT = LibDataTypes.TEXT;
-  type NUMBER = LibDataTypes.NUMBER;
-  type INTEGER = LibDataTypes.INTEGER;
-  type BIGINT = LibDataTypes.BIGINT;
-  type FLOAT = LibDataTypes.FLOAT;
-  type REAL = LibDataTypes.REAL;
-  type DOUBLE = LibDataTypes.DOUBLE;
-  type DECIMAL = LibDataTypes.DECIMAL;
-  type BOOLEAN = LibDataTypes.BOOLEAN;
-  type TIME = LibDataTypes.TIME;
-  type DATE = LibDataTypes.DATE;
-  type DATEONLY = LibDataTypes.DATEONLY;
-  type HSTORE = LibDataTypes.HSTORE;
-  type JSON = LibDataTypes.JSON;
-  type JSONB = LibDataTypes.JSONB;
-  type NOW = LibDataTypes.NOW;
-  type BLOB = LibDataTypes.BLOB;
-  type RANGE<T> = LibDataTypes.RANGE<T>;
-  type UUID = LibDataTypes.UUID;
-  type UUIDV1 = LibDataTypes.UUIDV1;
-  type UUIDV4 = LibDataTypes.UUIDV4;
-  type VIRTUAL<T> = LibDataTypes.VIRTUAL<T>;
-  type ENUM<T extends string> = LibDataTypes.ENUM<T>;
-  type ARRAY<T extends new () => ABSTRACT> = LibDataTypes.ARRAY<T>;
-  type GEOMETRY = LibDataTypes.GEOMETRY;
-  type GEOGRAPHY = LibDataTypes.GEOGRAPHY;
-  type DOUBLE_PRECISION = LibDataTypes.DOUBLE;
+
+  export type DataTypes = LibDataTypes;
```

Comment:
Btw, I wanted to point out that this and the `as LibDataTypes` is only needed because we can only export types and interfaces in this module, because the module is named like the `var` you define at the bottom. As soon as you export a `const` or a `class` you will get `duplicate identifier` error. This will impose a problem sooner or later if Sequelize 4.0 uses real classes... Wrapping everything in that `sequelize` internal namespace shouldn't be needed actually, we could just remove it because typings will wrap everything in an external module.